File: Excel/800-899/855/855 Challenge.R
library(tidyverse)
library(readxl)

path <- "Excel/800-899/855/855 Champion Non Continuous.xlsx"
input <- read_excel(path, range = "A2:B24")
test <- read_excel(path, range = "D2:E6")

result = input %>%
  mutate(n = n(), .by = Champion) %>%
  mutate(
    Consecutive = row_number() != 1 & lag(Champion) == Champion
  ) %>%
  filter(
    max(Consecutive) == FALSE,
    n >= 2,
    .by = Champion
  ) %>%
  summarise(
    Years = str_c(Year, collapse = ", "),
    .by = Champion
  ) %>%
  arrange(Champion)

all.equal(result, test, check.attributes = FALSE)
# [1] TRUE

File: Excel/800-899/855/855 Challenge.py
import pandas as pd

path = "Excel/800-899/855/855 Champion Non Continuous.xlsx"
input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=23)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=4)

input['n'] = input.groupby('Champion')['Champion'].transform('size')
input['Consecutive'] = (input['Champion'] == input['Champion'].shift()) & (input.index != 0)
valid = (~input.groupby('Champion')['Consecutive'].transform('max')) & (input['n'] >= 2)
result = (
    input[valid]
    .groupby('Champion', as_index=False)
    .agg({'Year': lambda x: ', '.join(map(str, x))})
    .sort_values('Champion')
    .rename(columns={'Champion': 'Country', 'Year': 'Years'})
)

print(result.equals(test)) # True
